### DISCUSSION TITLE
Reverse Automatic Mapping

### DISCUSSION THREAD

**Discussion by rlopez-plenty** [cuelang/cue#900] [Q&A]
Hello Community!
We want to create a way to take all the mappings between the different objects we have, and automatically generate the inverse mapping of each one. Here is some code and some thoughts about how we think it can work but we still haven't found a way to handle it. Do you have any idea how to resolve this with Cuelang, following this approach or any other idea is welcome. Thanks! (We are using 0.2.2 version)

```
// This final output matches what we are hoping to achieve
// but is hardcoded to work, need a dynamic solution
TEMP: {
    // If we define a mapping here, the inverse should show up at that mapping
    // ie, Alpha.Alpha1.id should now map to Beta.Beta1.id
    // (are there potential recursive infinite loop problems?)
    #Object: {...}
    Beta: #Object & {
	Beta1: #Object & {
		id: "B1234"
		mappings: [{to: Alpha.Alpha1.id}, {to: Alpha.Alpha1.id}]
	}
    }
    // This code chunk would ideally live under Beta1 or Beta1.mappings
    let temp = Beta.Beta1.mappings
    for x in temp {
    	// Ideally we would have access to Alpha.Alpha1.id structure here
    	// like "\(x.something)" = Alpha, "\(x.something.else)" = Alpha1
    	Alpha: {
    		Alpha1: {
			mappings: [{to: Beta.Beta1.id}]
        	}
    	}
    }
    // Object we want to map back to Beta.Beta1.id
    Alpha: #Object & {
	Alpha1: #Object & {
		id:    "A1234"
		mappings: [{to: Beta.Beta1.id}]
	}
    }
}
```

Here is the sample data:
```
// This final output matches what we are hoping to achieve
// but is hardcoded to work, need a dynamic solution
TEMP: {
	Beta: {
		Beta1: {
			id: "B1234"
			mappings: [{
				to: "A1234"
			}, {
				to: "A1234"
			}]
		}
	}
	// Object we want to map back to Beta.Beta1.id
	Alpha: {
		Alpha1: {
			id: "A1234"
			mappings: [{
				to: "B1234"
			}]
		}
	}
	// Ideally we would have access to Alpha.Alpha1.id structure here
	// like "\(x.something)" = Alpha, "\(x.something.else)" = Alpha1
	r: {
		to: "A1234"
	}
}

```

**Note: Indentation is not working properly, you can assume it has just one tab per line.**

**Comment by mpvl** [comment]
Could you give an example in terms of data only?

The indentation seems somewhat off, so not sure what is intended.

**Reply by rlopez-plenty** [reply]
Sure, of course, I already changed the description. Thanks! @mpvl Can you please take a look? Thanks in advance!
I created this discussion to clarify the issue we are having: 
https://github.com/cuelang/cue/discussions/907

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/900.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
